kratos/python_scripts/time_based_ascii_file_writer_utility.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import print_function, absolute_import, division  # makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+
+# Importing the Kratos Library
+import KratosMultiphysics
+
+# Other imports
+import os
+
+class TimeBasedAsciiFileWriterUtility(object):
+    """This utility handles a file to which results are to be written.
+    It internally handles restarts and the closing/opening of files.
+
+    It should be used by e.g. processes that write data to files
+    """
+    def __init__(self, model_part, params, file_header):
+
+        default_settings = KratosMultiphysics.Parameters('''{
+            "output_file_name"  : "",
+            "write_buffer_size" : -1
+        }''')
+
+        self.model_part = model_part
+
+        params.ValidateAndAssignDefaults(default_settings)
+
+        self.output_file_name = params["output_file_name"].GetString()
+
+        # size of the buffer in bytes. Set to "0" for flushing always
+        self.write_buffer_size = params["write_buffer_size"].GetInt()
+
+        if not self.model_part.ProcessInfo[KratosMultiphysics.IS_RESTARTED]:
+            self.file = self.__InitializeOutputFile(file_header)
+        else:
+            restart_time = self.model_part.ProcessInfo[KratosMultiphysics.TIME]
+            existing_file_is_valid, out_file = self.__AddToExistingOutputFile(file_header, restart_time)
+
+            if existing_file_is_valid:
+                self.file = out_file
+            else:
+                warn_msg  = "No valid data file was found after restarting,\n"
+                warn_msg += "writing to a new file"
+                KratosMultiphysics.Logger.PrintWarning("TimeBasedAsciiFileWriterUtility", warn_msg)
+                self.file = self.__InitializeOutputFile(file_header)
+
+    def write(self, string_to_write):
+        self.file.write(string_to_write)
+
+    def flush(self):
+        self.file.flush()
+
+    def close(self):
+        if not self.file.closed:
+            self.file.close()
+
+    def __OpenOutputFile(self):
+        if self.write_buffer_size == -1: # user didn't specify a buffer size, -1 means we use the system default
+            return open(self.output_file_name,"w")
+        else:
+            return open(self.output_file_name,"w", buffering=self.write_buffer_size)
+
+    def __InitializeOutputFile(self, file_header):
+        output_file = self.__OpenOutputFile()
+        output_file.write(file_header)
+        return output_file
+
+    def __AddToExistingOutputFile(self, file_header, restart_time):
+        if not os.path.isfile(self.output_file_name):
```

Comment:
Why not just return `None`? One return value is easier to understand. One can check if the file is valid with `if my_file:`.